packages/store/addon/-private/system/record-array-manager.js

```diff
@@ -8,347 +8,804 @@ import { get, set } from '@ember/object';
 import { assign } from '@ember/polyfills';
 import { run as emberRunloop } from '@ember/runloop';
 
+import { RECORD_ARRAY_MANAGER_IDENTIFIERS, RECORD_ARRAY_MANAGER_LEGACY_COMPAT } from '@ember-data/canary-features';
+
+// only used by RECORD_ARRAY_MANAGER_IDENTIFIERS
+import isStableIdentifier from '../identifiers/is-stable-identifier';
 import { AdapterPopulatedRecordArray, RecordArray } from './record-arrays';
 import { internalModelFactoryFor } from './store/internal-model-factory';
 
+const RecordArraysCache = new WeakMap();
 const emberRun = emberRunloop.backburner;
+let RecordArrayManager;
+
+export let associateWithRecordArray;
```

Comment:
Lets export this after it's been assigned

Reply:
Since we have this assigned in a conditional block, it only allows us to import/export at the top level.